### ISSUE TITLE
conditional sampling

### ISSUE THREAD

**Issue by Dashi-1997** [kamenbliznashki/pixel_models#2]
Hello, Can your code implement conditional sampling?

**Comment by kamenbliznashki** [comment]
Yes, all the models can be conditional. I used conditional PixelCNN and
PixelSNAIL in the VQVAE2 implementation in my generative models repo here
<https://github.com/kamenbliznashki/generative_models>, where the pixel
models are used as priors and sampled from conditionally. Take a look at
vqvae_prior.py

**Comment by Dashi-1997** [comment]
Thanks for your reply! I am wondering about how to use conditions in VQ-VAE and I will read your code carefully.

**Comment by liyikang0613** [comment]
Hi, Can you please tell me how to train the Pixel CNN++ on other datasets besides the three datasets you `provided? This is my set data function but I have no idea how to add it to the model. Could you please help me with it? Thank you.
[setdata.zip](https://github.com/kamenbliznashki/pixel_models/files/5397630/setdata.zip)

**Comment by kamenbliznashki** [comment]
Hi - the best way to include new datasets is to update `fetch_dataloaders` in `main.py` for the dataset and dataloaders you'd like to train on.
